File: leetcode/AIcoding/3.DeepSeek-v3/c_solutions/553-optimal-division.c
char* optimalDivision(int* nums, int numsSize) {
    if (numsSize == 1) {
        char* result = malloc(12);
        sprintf(result, "%d", nums[0]);
        return result;
    }
    if (numsSize == 2) {
        char* result = malloc(24);
        sprintf(result, "%d/%d", nums[0], nums[1]);
        return result;
    }

    int length = 0;
    for (int i = 0; i < numsSize; i++) {
        int num = nums[i];
        if (num < 0) num = -num;
        while (num > 0) {
            length++;
            num /= 10;
        }
        length++;
    }
    length += numsSize * 2;

    char* result = malloc(length);
    int pos = 0;

    pos += sprintf(result + pos, "%d/(", nums[0]);
    for (int i = 1; i < numsSize - 1; i++) {
        pos += sprintf(result + pos, "%d/", nums[i]);
    }
    pos += sprintf(result + pos, "%d)", nums[numsSize - 1]);

    return result;
}

File: leetcode/AIcoding/1.Gemini-2.5/python3_solutions/553-optimal-division.py
class Solution:
    def optimalDivision(self, nums: list[int]) -> str:
        n = len(nums)
        if n == 1:
            return str(nums[0])
        elif n == 2:
            return str(nums[0]) + "/" + str(nums[1])
        else:
            result = [str(nums[0]), "/("]
            for i in range(1, n):
                result.append(str(nums[i]))
                if i < n - 1:
                    result.append("/")
            result.append(")")
            return "".join(result)